### PR TITLE
Add functionality to ScheduleBlockedFault

### LINKED ISSUE #293: Add functionality to ScheduleBlockedFault
## User stories
As a developer I want the ScheduleBlockedFault to call the Spawner in order to block schedules.

## Acceptance Criteria
- [x] implement inject_fault method
- [x] implement resolve_fault method
- [x] log events
- [x] test ScheduleBlockedFault

### PR DESCRIPTION
**Pull request by instant-sky**
Fixes #293 

Description (what might a PO or reviewer want to know?)

## PR checklist

- [x] Acceptance criteria fulfilled
- [x] Additional features are tested
- [ ] Docs (code, wiki & diagrams) updated
- [ ] Breaking changes anounced
- [x] Dev-branch has been merged into local branch to resolve conflicts
- [x] Tests and linter have passed AFTER local merge
- [x] Another dev reviewed and approved

### CODE REVIEW

File: src/fault_injector/fault_types/schedule_blocked_fault.py [hunk 1]
```diff
@@ -1,23 +1,48 @@
...
         :param tick: the simulation tick in which inject_fault was called
         :type tick: Integer
         """
+        self.spawner.block_schedule(self.configuration.affected_element_id)
+        self.logger.inject_schedule_blocked_fault(
+            tick=tick,
+            schedule_blocked_fault_configuration=self.configuration.id,
+            affected_element=self.configuration.affected_element_id,
+        )
         # - get schedule by id
```
Comment (antonykamp) [inline src/fault_injector/fault_types/schedule_blocked_fault.py]: Do you still need the comments? 🤔

File: tests/fault_injector/test_schedule_blocked_fault.py [hunk 2]
```diff
@@ -0,0 +1,106 @@
...
+        return ScheduleBlockedFault(
+            configuration=schedule_blocked_fault_configuration,
+            logger=logger,
+            spawner=spawner,
+        )
+
+    # It would be better to test if trains spawn after inject_fault.
+    # As of now this is not really possible, the tests should therefore edited in the future
+    # pylint: disable=protected-access
+    def test_inject_schedule_blocked_fault(
```
Comment (antonykamp) [inline tests/fault_injector/test_schedule_blocked_fault.py:89]: Would you like to test the logging statements too? 😇

**Review by antonykamp — COMMENTED** [review]
Looks good 👍 
I just have two minor questions 😇 Don't worry ;)

File: tests/fault_injector/test_schedule_blocked_fault.py (hunk 2 above)
Comment (instant-sky) [reply tests/fault_injector/test_schedule_blocked_fault.py:89]: Yes, I discussed this already with @LucasDerReisende and we decided to do this with an extra issue

File: src/fault_injector/fault_types/schedule_blocked_fault.py (hunk 1 above)
Comment (instant-sky) [reply src/fault_injector/fault_types/schedule_blocked_fault.py]: Good point, that should be documented by the code now. 📜

Comment (antonykamp) [reply src/fault_injector/fault_types/schedule_blocked_fault.py]: Thanks! 💚

File: tests/fault_injector/test_schedule_blocked_fault.py (hunk 2 above)
Comment (antonykamp) [reply tests/fault_injector/test_schedule_blocked_fault.py:89]: Okedoke 👍

**Review by antonykamp — APPROVED** [review]

**Pull request merged**